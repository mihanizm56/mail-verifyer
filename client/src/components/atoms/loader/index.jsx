import React, { memo } from "react";
import { SVGIcon } from "../svg-icon";
import loaderIcon from "../../../assets/icons/icon-loading.svg";
import "../../../styles/keyframes.css";
import "./loader.css";

export const Loader = memo(() => (
	<div className="loader">
		<SVGIcon path={loaderIcon} className="loader" />
	</div>
));
