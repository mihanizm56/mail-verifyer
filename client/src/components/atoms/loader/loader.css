.loader {
    height: 150px;
    width: 150px;
    /* border: 1px solid red; */
    animation: circleLoadingAnimation 2s linear infinite;
}