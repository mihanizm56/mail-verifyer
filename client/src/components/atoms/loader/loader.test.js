import React, { createElement } from "react";
import { shallow } from "enzyme";
import { Loader } from "../loader";

describe("test Loader", () => {
	beforeEach(() => {
		jest.resetModules();
		jest.clearAllMocks();
	});

	describe("test render contract", () => {
		it("render snap with all props", () => {
			const component = shallow(<Loader />);

			expect(component).toMatchSnapshot();
		});
	});
});
