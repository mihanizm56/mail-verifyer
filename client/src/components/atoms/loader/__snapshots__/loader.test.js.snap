// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button test render contract render snap with all props 1`] = `
<div
  className="loader"
>
  <Component
    className="loader"
    path="icon-loading.svg"
  />
</div>
`;

exports[`test Loader test render contract render snap with all props 1`] = `
<div
  className="loader"
>
  <Component
    className="loader"
    path="icon-loading.svg"
  />
</div>
`;
